src/executables/master_heartbeat.c

```diff
@@ -2658,13 +2658,14 @@ hb_cluster_load_group_and_node_list (char *ha_node_list, char *ha_replica_list)
 {
   int priority, num_nodes;
   char tmp_string[LINE_MAX];
+  char err_string[LINE_MAX];
   char *p, *savep;
   HB_NODE_ENTRY *node;
 
-  if (ha_node_list == NULL)
+  if (ha_node_list == NULL || ha_node_list[0] == '\0')
     {
-      MASTER_ER_LOG_DEBUG (ARG_FILE_LINE, "invalid ha_node_list. (ha_node_list:NULL).\n");
-      return ER_FAILED;
+      sprintf (err_string, "%s is empty", prm_get_name (PRM_ID_HA_NODE_LIST));
```

Comment:
I recommend using the snprintf function.

Reply:
thank you, I will modify it as your recommand.